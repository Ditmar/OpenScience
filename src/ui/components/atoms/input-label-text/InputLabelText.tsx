import { Typography } from '@mui/material';
import type { IProps } from './types/IProps';
import { textSx } from './InputLabelText.styles';

export function InputLabelText(props: IProps) {
  const { label, error, disabled, helper, size } = props;

  const labelStyles = textSx({ disabled, error, helper, size });

  return (
    <Typography variant="body1" sx={labelStyles}>
      {label || ''}
    </Typography>
  );
}
