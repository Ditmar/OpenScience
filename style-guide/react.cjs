const { resolve } = require('node:path');
const project = resolve(process.cwd(), 'tsconfig.json');

module.exports = {
  env: {
    browser: true,
    es2021: true,
    es6: true,
    jest: true,
  },
  extends: [
    require.resolve('./rules/base'),
    require.resolve('./configurations/_base'),
    'plugin:react-hooks/recommended',
    'airbnb',
    'plugin:storybook/recommended',
    require.resolve('./configurations/typescript'),
    require.resolve('./configurations/vitest'),
  ],
  parserOptions: {
    project,
    babelOptions: {
      presets: ['@babel/preset-react'],
    },
  },
  plugins: ['react-refresh'],
  settings: {
    react: {
      version: 'detect',
    },
  },
  globals: {
    JSX: true,
  },
  rules: {
    'react/react-in-jsx-scope': 'off',
    'react/button-has-type': 'off',
    'react-refresh/only-export-components': [
      'warn',
      { allowExportNames: ['meta', 'links', 'headers', 'loader', 'action'] },
    ],
  },
};
