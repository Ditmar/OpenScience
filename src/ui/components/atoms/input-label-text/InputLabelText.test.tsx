import { render } from '@testing-library/react';
import { InputLabelText } from './InputLabelText';

describe('InputLabelText Component', () => {
  it('renders disabled state', () => {
    const { container } = render(<InputLabelText label="Input Label" disabled />);
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot with error', () => {
    const { container } = render(<InputLabelText label="Input Label" error />);
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot disabled', () => {
    const { container } = render(<InputLabelText label="Input Label" disabled />);
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot helper', () => {
    const { container } = render(<InputLabelText label="Input Label" helper />);
    expect(container).toMatchSnapshot();
  });
});
