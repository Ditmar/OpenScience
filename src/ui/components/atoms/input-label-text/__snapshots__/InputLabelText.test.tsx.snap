// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InputLabelText Component > matches snapshot disabled 1`] = `
<div>
  <p
    class="MuiTypography-root MuiTypography-body1 css-1b4s42d-MuiTypography-root"
  >
    Input Label
  </p>
</div>
`;

exports[`InputLabelText Component > matches snapshot helper 1`] = `
<div>
  <p
    class="MuiTypography-root MuiTypography-body1 css-9lmazj-MuiTypography-root"
  >
    Input Label
  </p>
</div>
`;

exports[`InputLabelText Component > matches snapshot with error 1`] = `
<div>
  <p
    class="MuiTypography-root MuiTypography-body1 css-ek0aq9-MuiTypography-root"
  >
    Input Label
  </p>
</div>
`;

exports[`InputLabelText Component > renders disabled state 1`] = `
<div>
  <p
    class="MuiTypography-root MuiTypography-body1 css-1b4s42d-MuiTypography-root"
  >
    Input Label
  </p>
</div>
`;
